src/github/client/interceptors.ts

```diff
@@ -84,10 +85,10 @@ export const handleFailedRequest = (logger: Logger) =>
 			const errorMessage = `Error executing Axios Request ` + error.message;
 
 			const rateLimitResetHeaderValue:string = response.headers?.["x-ratelimit-reset"];
-			const rateLimitRemainingHeaderValue:string = response.headers?.["x-ratelimit-reset"];
-			if(response.headers?.["x-ratelimit-remaining"] == "0" && rateLimitResetHeaderValue) {
+			const rateLimitRemainingHeaderValue:string = response.headers?.["x-ratelimit-remaining"];
+			if(status === 403 && rateLimitRemainingHeaderValue == "0") {
```

Comment:
We should rely on GitHub's 403 status to throw a `RateLimitError`. Other requests with ratelimit-remaining==0 may go through fine (it's just the last request that will go through).

Also, there was a copy&paste error with the `x-ratelimit-remaining` header name

Reply:
Was there a case where we had 403 but no x-ratelimit-remaining header, but it was rate limiting?